plugins/modules/aci_dhcp_relay_provider.py

```diff
@@ -330,28 +396,46 @@ def main():
         tdn = "uni/tn-{0}/out-{1}/instP-{2}".format(provider_tenant, l3out_name, external_epg)
     elif epg_type == "dn":
         tdn = dn
+    else:
+        tdn = None
 
-    aci = ACIModule(module)
-    aci.construct_url(
-        root_class=dict(
+    if tenant is None:
+        root_class = dict(
```

Comment:
why recreating the whole structure and not just changing the root class part to be class infra / rn infra / ... ?

Reply:
I am not recreating the structure, I am creating the correct structure based on a conditional. The reason why I chose to do it this way was to simplify the logic, because the order of classes is different.

1. infra/dhcpRelayP, dhcpRsProv, None ( no subclass 2 needed )
2. fvTenant, dhcpRelayP, dhcpRsProv

I also tried the option of only setting root class to infra, but this impacts the output for query_all. Where infra, with children would be displayed. This deviates with other modules query_all behaviour.

It might be possible with more conditional logic to construct it, but seemed more complicated approach.